src/core/gcc.rs

```diff
@@ -214,6 +229,12 @@ pub fn client_core_data(parameter: Option<ClientData>) -> Component {
         client_parameter.name.clone() + &"\x00".repeat(16 - client_parameter.name.len())
     };
 
+    let mut capability_flags = CapabilityFlag::RnsUdCsSupportErrinfoPDU as u16;
+    if let Some(_) = client_parameter.connection_type {
+        // indicate that we're specifying the connection type
+        capability_flags |= CapabilityFlag::RnsUdCsValidConnectionType as u16;
+    }
```

Comment:
nit: we generally prefer using matches! or is_some when we don't need the inner value

```suggestion
    if client_parameter.connection_type.is_some() {
        // indicate that we're specifying the connection type
        capability_flags |= CapabilityFlag::RnsUdCsValidConnectionType as u16;
    }
```